src/routes/boot.ts

```diff
@@ -95,8 +95,8 @@ export type BootUserReferral = Partial<{
 }>;
 
 interface AnonymousUser extends BootUserReferral {
-  id: string;
-  firstVisit: string;
+  id?: string;
+  firstVisit: string | null;
```

Comment:
Why not?
```suggestion
  firstVisit?: string;
```

Reply:
mostly because underlying typeorm and redis methods return null instead of undefined and i did not want to make more type errors while I change those. Usually yes, we want undefined (optional).

Reply:
this applies to all comments above

Reply:
also null and undefined are different things so TS complains about it, you cant use it interchangeably

Reply:
Ah ok, yeah figured it was related to something like that.
Ok I'll resolve all above :)